docs/extending_deptrac.md

```diff
@@ -0,0 +1,96 @@
+# Extending Deptrac
+
+Deptrac defines its extension point by providing set of **contract** classes
+that you can depend on for your implementation. The classes can be found in
+the `src/Contract` directory and are covered by
+the [backwards compatibility policy](bc_policy.md).
+
+There are several ways you can extend Deptrac:
+
+- [Output formatters](#Output Formatters) - To change how the output of
+  dependency analysis is formatted. This might be useful to integrate Deptrac
+  into your CI/CD pipeline.
+- [Analyser Events](#Analyser Events) - Decide whether a dependency is
+  Uncovered, Allowed, Violation or SkippedViolation.
+- [Layer collectors](#Layer collectors) - Add custom deptrac collector
+
+> **Note**
+> In examples where FQCN is not specified, the base
+> namespace `Qossmic\Deptrac\Contract\` is omitted for readability.
+
+## Output Formatters
+
+Creating an output formatter requires only to create a new class implementing
+the `OutputFormatter\OutputFormatterInterface` and
+register it
+in your `deptrac.yaml` file like this:
+
+```yaml
+services:
+  - class: <Your FQCN>
+    autowire: true
+    tags:
+      - output_formatter
+```
+
+And you are done. You can call your formatter by using the `-f` or `--formatter`
+CLI flag with the name you defined
+in `OutputFormatter\OutputFormatterInterface::getName()`.
+
+## Analyser Events
```

Comment:
It would be nice to have a tabular overview, which events Deptrac emits and what data it contains, but we can do this in a follow up.

Reply:
Right now it is only 2 events that are exposed in the contract. I don't think it currently warrants a table.